neo4j/session_with_context_test.go

```diff
@@ -569,7 +569,74 @@ func TestSession(st *testing.T) {
 		})
 	})
 
-	st.Run("Close", func(ct *testing.T) {
+	outer.Run("GetServerInfo", func(inner *testing.T) {
+
+		inner.Run("Retrieves info from first successful connection", func(t *testing.T) {
```

Comment:
It's not really testing it it's the first successful connection as the fake pool will return the same fake connection as many times as asked to.

Reply:
I can make the test a bit stricter by introducing `BorrowHook`